tests/ui/codegen/nondetermenistic-int-expose-provenance.rs

```diff
@@ -0,0 +1,21 @@
+//@ run-pass
+//@ known-bug: #107975
+
+#![feature(exposed_provenance)]
+
+use std::ptr::addr_of;
+
+fn main() {
+    let a: usize = {
+        let v = 0u8;
+        addr_of!(v).expose_provenance()
+    };
+    let b: usize = {
+        let v = 0u8;
+        addr_of!(v).expose_provenance()
+    };
+    let i: usize = a - b;
+    assert_ne!(i, 0);
+    println!("{}", i);
+    assert_eq!(i, 0);
+}
```

Comment:
Maybe add some more scenarios like https://github.com/rust-lang/rust/issues/107975#issuecomment-1430704499, https://github.com/rust-lang/rust/issues/107975#issuecomment-1431758601, https://github.com/rust-lang/rust/issues/107975#issuecomment-1434203908 and https://github.com/rust-lang/rust/issues/107975#issuecomment-1432161340

Reply:
Done